go/service/chat_local.go

```diff
@@ -130,17 +210,76 @@ getinbox:
 		}
 	}
 
-	id, err := h.NewConversationLocal(ctx, chat1.ConversationIDTriple{
-		Tlfid:     tlfID,
-		TopicType: arg.TopicType,
-		// TopicID filled by server?
-	})
-	if err != nil {
-		return ids, err
+	return conversations, nil
+}
+
+// getConversationInfo locates the conversation by using id, and returns with
+// all fields filled in conversationInfo, along with a ConversationIDTriple
+//
+// TODO: cache
+func (h *chatLocalHandler) getConversationInfo(ctx context.Context, id chat1.ConversationID) (conversationInfo keybase1.ConversationInfoLocal, triple chat1.ConversationIDTriple, err error) {
+	ipagination := &chat1.Pagination{Num: 20}
+getinbox:
+	for i := 0; i < 10000; /* in case we have a server bug */ i++ {
+		iview, err := h.GetInboxLocal(ctx, ipagination)
+		if err != nil {
+			return conversationInfo, triple, err
+		}
+
+		for _, conversation := range iview.Conversations {
+			if id == conversation.Metadata.ConversationID {
+				triple = conversation.Metadata.IdTriple
+				conversationInfo.Id = id
+				conversationInfo.TopicType = triple.TopicType
+
+				if len(conversation.MaxHeaders) == 0 {
+					return conversationInfo, triple, errors.New("empty conversation found")
```

Comment:
That's an interesting idea! I like not exposing TlfID at all. My concern is it's still not signed, and we could potentially make the mistake of using this information when it's contradictory (presumably caused by a rogue server) with signed information in message metadata. Also two more thoughts:

1) we are including TlfName in every message. If we move this (and the ConversationIDTriple, for the same sake) to conversation metadata and only include a hash of the metadata, the client header can be smaller. Is there any downsides of only signing hashes? (I vaguely remember we went over this before, but can't remember the rationale now)

2) If we go with the [relying on message metadata] approach, it looks like we can remove the ConversationIDTriple from the result of GetInboxView all together. If the triple is needed, the client can get it from MessageClientHeader, which is signed.